libraries/standard/http/utest/http_utest.c

```diff
@@ -248,4 +287,190 @@ void test_Http_InitializeRequestHeaders_insufficient_memory()
                                HTTP_TEST_REQUEST_LINE_LEN ) != 0 );
 }
 
+/* ===================== Testing HTTPClient_AddHeader ======================= */
+
+/**
+ * @brief Prefill the user buffer with HTTP_TEST_HEADER_REQUEST_LINE and call
+ * HTTPClient_AddHeader using HTTP_TEST_HEADER_FIELD and HTTP_TEST_HEADER_VALUE.
+ */
+void test_Http_AddHeader_happy_path()
+{
+    HTTPStatus_t test_err = HTTP_INTERNAL_ERROR;
+    HTTPRequestHeaders_t requestHeaders = { 0 };
+    size_t expectedHeaderLen = HTTP_TEST_SINGLE_HEADER_LEN;
+    char expectedHeader[ HTTP_TEST_DOUBLE_HEADER_BUFFER_LEN ] = { 0 };
+    int numBytes = 0;
+
+    setupBuffer( &requestHeaders, HTTP_TEST_DOUBLE_HEADER_LEN );
+
+    /* Add 1 because snprintf(...) writes a null byte at the end. */
+    numBytes = snprintf( expectedHeader, HTTP_TEST_SINGLE_HEADER_LEN + 1,
+                         HTTP_TEST_SINGLE_HEADER_FORMAT,
+                         HTTP_TEST_HEADER_REQUEST_LINE,
+                         HTTP_TEST_HEADER_FIELD, HTTP_TEST_HEADER_VALUE );
+    TEST_ASSERT_EQUAL( numBytes, HTTP_TEST_SINGLE_HEADER_LEN );
+    /* Set parameters for requestHeaders. */
+    numBytes = snprintf( ( char * ) requestHeaders.pBuffer,
+                         HTTP_TEST_HEADER_REQUEST_LINE_LEN + 1,
+                         HTTP_TEST_HEADER_REQUEST_LINE );
+    TEST_ASSERT_EQUAL( numBytes, HTTP_TEST_HEADER_REQUEST_LINE_LEN );
+    /* We correctly set headersLen after writing request line to buffer. */
+    requestHeaders.headersLen = HTTP_TEST_HEADER_REQUEST_LINE_LEN;
+    test_err = HTTPClient_AddHeader( &requestHeaders,
+                                     HTTP_TEST_HEADER_FIELD, HTTP_TEST_HEADER_FIELD_LEN,
+                                     HTTP_TEST_HEADER_VALUE, HTTP_TEST_HEADER_VALUE_LEN );
+    TEST_ASSERT_EQUAL_MEMORY( requestHeaders.pBuffer,
+                              expectedHeader, expectedHeaderLen );
+    TEST_ASSERT_EQUAL( requestHeaders.headersLen, expectedHeaderLen );
```

Comment:
Just realized also that expected comes first then the actual. Could you please switch these everywhere so they get printed correctly when something goes wring?